services/workerplane.go

```diff
@@ -4,43 +4,38 @@ import (
 	"github.com/rancher/rke/hosts"
 	"github.com/rancher/types/apis/management.cattle.io/v3"
 	"github.com/sirupsen/logrus"
+	"golang.org/x/sync/errgroup"
 )
 
 func RunWorkerPlane(controlHosts, workerHosts []*hosts.Host, workerServices v3.RKEConfigServices, nginxProxyImage, sidekickImage string, healthcheckDialerFactory hosts.DialerFactory) error {
 	logrus.Infof("[%s] Building up Worker Plane..", WorkerRole)
+	var errgrp errgroup.Group
+
+	// Deploy worker components on control hosts
 	for _, host := range controlHosts {
-		// run sidekick
-		if err := runSidekick(host, sidekickImage); err != nil {
-			return err
-		}
-		// run kubelet
-		// only one master for now
-		if err := runKubelet(host, workerServices.Kubelet, healthcheckDialerFactory); err != nil {
-			return err
-		}
-		if err := runKubeproxy(host, workerServices.Kubeproxy, healthcheckDialerFactory); err != nil {
-			return err
-		}
+		host := host
```

Comment:
Why the reassignment here ?

Reply:
this is related to this https://golang.org/doc/faq#closures_and_goroutines